<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QgsModelDesignerDialogBase</class>
 <widget class="QMainWindow" name="QgsModelDesignerDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>786</width>
    <height>596</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Model Designer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">
    <property name="spacing">
     <number>3</number>
    </property>
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item>
     <widget class="QgsModelGraphicsView" name="mView"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="mMenu">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>786</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_Model">
    <property name="title">
     <string>&amp;Model</string>
    </property>
    <widget class="QMenu" name="menuExport">
     <property name="title">
      <string>Export</string>
     </property>
     <addaction name="mActionExportImage"/>
     <addaction name="mActionExportPdf"/>
     <addaction name="mActionExportSvg"/>
     <addaction name="separator"/>
     <addaction name="mActionExportPython"/>
    </widget>
    <addaction name="mActionValidate"/>
    <addaction name="mActionRun"/>
    <addaction name="mActionRunSelectedSteps"/>
    <addaction name="separator"/>
    <addaction name="mActionReorderInputs"/>
    <addaction name="mActionReorderOutputs"/>
    <addaction name="separator"/>
    <addaction name="mActionNew"/>
    <addaction name="mActionOpen"/>
    <addaction name="mActionSave"/>
    <addaction name="mActionSaveAs"/>
    <addaction name="mActionSaveInProject"/>
    <addaction name="mActionEditHelp"/>
    <addaction name="separator"/>
    <addaction name="menuExport"/>
    <addaction name="separator"/>
    <addaction name="mActionClose"/>
   </widget>
   <widget class="QMenu" name="mMenuView">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="mActionZoomIn"/>
    <addaction name="mActionZoomOut"/>
    <addaction name="mActionZoomActual"/>
    <addaction name="mActionZoomToItems"/>
    <addaction name="separator"/>
    <addaction name="mActionShowComments"/>
    <addaction name="separator"/>
    <addaction name="mActionSnappingEnabled"/>
    <addaction name="separator"/>
    <addaction name="mActionHidePanels"/>
   </widget>
   <widget class="QMenu" name="mMenuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="mActionSelectAll"/>
    <addaction name="mActionSnapSelected"/>
    <addaction name="separator"/>
    <addaction name="mActionDeleteComponents"/>
    <addaction name="separator"/>
    <addaction name="mActionAddGroupBox"/>
   </widget>
   <addaction name="menu_Model"/>
   <addaction name="mMenuEdit"/>
   <addaction name="mMenuView"/>
  </widget>
  <widget class="QToolBar" name="mToolbar">
   <property name="windowTitle">
    <string>Navigation</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="mActionNew"/>
   <addaction name="mActionOpen"/>
   <addaction name="mActionSave"/>
   <addaction name="mActionSaveAs"/>
   <addaction name="mActionSaveInProject"/>
   <addaction name="separator"/>
   <addaction name="mActionPan"/>
   <addaction name="mActionSelectMoveItem"/>
   <addaction name="separator"/>
   <addaction name="mActionZoomIn"/>
   <addaction name="mActionZoomOut"/>
   <addaction name="mActionZoomActual"/>
   <addaction name="mActionZoomToItems"/>
   <addaction name="separator"/>
   <addaction name="mActionExportImage"/>
   <addaction name="mActionExportPdf"/>
   <addaction name="mActionExportSvg"/>
   <addaction name="separator"/>
   <addaction name="mActionEditHelp"/>
   <addaction name="separator"/>
   <addaction name="mActionRun"/>
  </widget>
  <widget class="QgsDockWidget" name="mPropertiesDock">
   <property name="windowTitle">
    <string>Model Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QgsScrollArea" name="scrollArea">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>256</width>
          <height>90</height>
         </rect>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="1">
          <widget class="QLineEdit" name="mNameEdit">
           <property name="toolTip">
            <string>Enter model name here</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Name</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Group</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="mGroupEdit">
           <property name="toolTip">
            <string>Enter group name here</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QgsDockWidget" name="mInputsDock">
   <property name="windowTitle">
    <string>Inputs</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QgsScrollArea" name="mInputsScrollArea">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>256</width>
          <height>173</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QgsModelDesignerInputsTreeWidget" name="mInputsTreeWidget">
           <column>
            <property name="text">
             <string notr="true">1</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="leftMargin">
            <number>6</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>6</number>
           </property>
           <item>
            <widget class="QPushButton" name="mReorderInputsButton">
             <property name="text">
              <string>Reorder Model Inputsâ€¦</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QgsDockWidget" name="mAlgorithmsDock">
   <property name="windowTitle">
    <string>Toolbox</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QgsScrollArea" name="scrollArea_2">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents_3">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>256</width>
          <height>167</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QgsFilterLineEdit" name="mToolboxSearchEdit">
           <property name="toolTip">
            <string>Enter algorithm name to filter list</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QgsProcessingToolboxTreeView" name="mToolboxTree">
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QgsDockWidget" name="mVariablesDock">
   <property name="windowTitle">
    <string>Variables</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_4">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QgsVariableEditorWidget" name="mVariablesEditor" native="true"/>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="mActionOpen">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionFileOpen.svg</normaloff>:/images/themes/default/mActionFileOpen.svg</iconset>
   </property>
   <property name="text">
    <string>Open Modelâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Open model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="mActionSave">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionFileSave.svg</normaloff>:/images/themes/default/mActionFileSave.svg</iconset>
   </property>
   <property name="text">
    <string>Save Model</string>
   </property>
   <property name="toolTip">
    <string>Save model</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="mActionSaveAs">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionFileSaveAs.svg</normaloff>:/images/themes/default/mActionFileSaveAs.svg</iconset>
   </property>
   <property name="text">
    <string>Save Model asâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Save model as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="mActionZoomActual">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionZoomActual.svg</normaloff>:/images/themes/default/mActionZoomActual.svg</iconset>
   </property>
   <property name="text">
    <string>Zoom to &amp;100%</string>
   </property>
   <property name="toolTip">
    <string>Zoom to 100%</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="mActionZoomIn">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionZoomIn.svg</normaloff>:/images/themes/default/mActionZoomIn.svg</iconset>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="toolTip">
    <string>Zoom in</string>
   </property>
   <property name="shortcut">
    <string>Ctrl++</string>
   </property>
  </action>
  <action name="mActionZoomOut">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionZoomOut.svg</normaloff>:/images/themes/default/mActionZoomOut.svg</iconset>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="toolTip">
    <string>Zoom out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="mActionExportImage">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSaveMapAsImage.svg</normaloff>:/images/themes/default/mActionSaveMapAsImage.svg</iconset>
   </property>
   <property name="text">
    <string>Export as Imageâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Export as image</string>
   </property>
  </action>
  <action name="mActionZoomToItems">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionZoomFullExtent.svg</normaloff>:/images/themes/default/mActionZoomFullExtent.svg</iconset>
   </property>
   <property name="text">
    <string>Zoom Full</string>
   </property>
   <property name="toolTip">
    <string>Zoom full</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+0</string>
   </property>
  </action>
  <action name="mActionExportPdf">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSaveAsPDF.svg</normaloff>:/images/themes/default/mActionSaveAsPDF.svg</iconset>
   </property>
   <property name="text">
    <string>Export as PDFâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Export as PDF</string>
   </property>
  </action>
  <action name="mActionExportSvg">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSaveAsSVG.svg</normaloff>:/images/themes/default/mActionSaveAsSVG.svg</iconset>
   </property>
   <property name="text">
    <string>Export as SVGâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Export as SVG</string>
   </property>
  </action>
  <action name="mActionExportPython">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSaveAsPython.svg</normaloff>:/images/themes/default/mActionSaveAsPython.svg</iconset>
   </property>
   <property name="text">
    <string>Export as Python Scriptâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Export as Python Script</string>
   </property>
  </action>
  <action name="mActionEditHelp">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionEditHelpContent.svg</normaloff>:/images/themes/default/mActionEditHelpContent.svg</iconset>
   </property>
   <property name="text">
    <string>Edit Model Helpâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Edit model help</string>
   </property>
  </action>
  <action name="mActionRun">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionStart.svg</normaloff>:/images/themes/default/mActionStart.svg</iconset>
   </property>
   <property name="text">
    <string>Run Modelâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Run model</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="mActionSaveInProject">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mAddToProject.svg</normaloff>:/images/themes/default/mAddToProject.svg</iconset>
   </property>
   <property name="text">
    <string>Save Model in Project</string>
   </property>
   <property name="toolTip">
    <string>Save model in project</string>
   </property>
  </action>
  <action name="mActionClose">
   <property name="text">
    <string>Close</string>
   </property>
  </action>
  <action name="mActionShowComments">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Comments</string>
   </property>
  </action>
  <action name="mActionExportAsScriptAlgorithm">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSaveAsPython.svg</normaloff>:/images/themes/default/mActionSaveAsPython.svg</iconset>
   </property>
   <property name="text">
    <string>Export as Script Algorithmâ€¦</string>
   </property>
  </action>
  <action name="mActionDeleteComponents">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionDeleteSelected.svg</normaloff>:/images/themes/default/mActionDeleteSelected.svg</iconset>
   </property>
   <property name="text">
    <string>Delete Selected Components</string>
   </property>
   <property name="toolTip">
    <string>Delete selected model components</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="mActionSnappingEnabled">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Enable Snapping</string>
   </property>
  </action>
  <action name="mActionSnapSelected">
   <property name="text">
    <string>Snap Selected Components to Grid</string>
   </property>
  </action>
  <action name="mActionSelectAll">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSelectAll.svg</normaloff>:/images/themes/default/mActionSelectAll.svg</iconset>
   </property>
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="mActionAddGroupBox">
   <property name="text">
    <string>Add Group Box</string>
   </property>
  </action>
  <action name="mActionValidate">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mIconSuccess.svg</normaloff>:/images/themes/default/mIconSuccess.svg</iconset>
   </property>
   <property name="text">
    <string>Validate Model</string>
   </property>
  </action>
  <action name="mActionReorderInputs">
   <property name="text">
    <string>Reorder Model Inputsâ€¦</string>
   </property>
  </action>
  <action name="mActionHidePanels">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Toggle Panel &amp;Visibility</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Tab</string>
   </property>
  </action>
  <action name="mActionPan">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionPan.svg</normaloff>:/images/themes/default/mActionPan.svg</iconset>
   </property>
   <property name="text">
    <string>Pan Layout</string>
   </property>
   <property name="shortcut">
    <string>P</string>
   </property>
  </action>
  <action name="mActionSelectMoveItem">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionSelect.svg</normaloff>:/images/themes/default/mActionSelect.svg</iconset>
   </property>
   <property name="text">
    <string>Select/Move Item</string>
   </property>
   <property name="toolTip">
    <string>Select/Move item</string>
   </property>
   <property name="shortcut">
    <string>V</string>
   </property>
  </action>
  <action name="mActionNew">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionFileNew.svg</normaloff>:/images/themes/default/mActionFileNew.svg</iconset>
   </property>
   <property name="text">
    <string>New Modelâ€¦</string>
   </property>
  </action>
  <action name="mActionReorderOutputs">
   <property name="text">
    <string>Reorder Output Layers...</string>
   </property>
   <property name="toolTip">
    <string>Sets the order for adding layers generated by the model to projects</string>
   </property>
  </action>
  <action name="mActionRunSelectedSteps">
   <property name="icon">
    <iconset resource="../../../images/images.qrc">
     <normaloff>:/images/themes/default/mActionRunSelected.svg</normaloff>:/images/themes/default/mActionRunSelected.svg</iconset>
   </property>
   <property name="text">
    <string>Run Selected Stepsâ€¦</string>
   </property>
   <property name="toolTip">
    <string>Run only the selected steps in the model</string>
   </property>
   <property name="shortcut">
    <string>Shift+F5</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsModelGraphicsView</class>
   <extends>QGraphicsView</extends>
   <header>qgsmodelgraphicsview.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFilterLineEdit</class>
   <extends>QLineEdit</extends>
   <header>qgsfilterlineedit.h</header>
  </customwidget>
  <customwidget>
   <class>QgsProcessingToolboxTreeView</class>
   <extends>QTreeView</extends>
   <header>qgsprocessingtoolboxtreeview.h</header>
  </customwidget>
  <customwidget>
   <class>QgsVariableEditorWidget</class>
   <extends>QWidget</extends>
   <header>qgsvariableeditorwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsDockWidget</class>
   <extends>QDockWidget</extends>
   <header>qgsdockwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsScrollArea</class>
   <extends>QScrollArea</extends>
   <header>qgsscrollarea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QgsModelDesignerInputsTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>qgsmodeldesignerinputstreewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../images/images.qrc"/>
 </resources>
 <connections/>
</ui>
